src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -554,8 +554,7 @@ public function dumpFile($filename, $content)
             throw new IOException(sprintf('Failed to write file "%s".', $filename), 0, null, $filename);
         }
 
-        // Ignore for filesystems that do not support umask
-        @chmod($tmpFile, 0666);
+        @chmod($tmpFile, 0666 & ~umask());
```

Comment:
There is nothing I could adjust, because the Symfony unit test suite explicitly sets the umask to `0`, so the expected file permissions in the tests will always be `666` on both Unix and Windows.